recovery-contracts/RecoveryWallet.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity 0.8.10;
+import "./RecoveryForwarder.sol";
+import "./CloneFactory.sol";
+/**
+ *
+ * RecoveryWallet
+ * ============
+ *
+ * Basic singleSig wallet designed to recover funds.
+ *
+ */
+contract RecoveryWallet is CloneFactory {
+
+  // Public fields
+  address public immutable signer;
+  address public immutable forwarderImplementationAddress;
+
+  constructor (address _signer, address _forwarderImplementationAddress) {
+    signer = _signer;
+    forwarderImplementationAddress = _forwarderImplementationAddress;
+  }
+  /**
+   * Modifier that will execute internal code block only if the sender is an authorized signer on this wallet
+   */
+  modifier onlySigner {
+    require( signer == msg.sender, 'Non-signer in onlySigner method');
+    _;
+  }
+
+  /**
+   * Gets called when a transaction is received with ether and no data
+   */
+  receive() external payable {
```

Comment:
i think you need fallback function as well

Reply:
We can add it as it doesn't increase gas usage however I think fallback function is not mandatory, it is only invoked when the data doesn't match any function

Reply:
added